relay-ourlogs/src/vercel_to_sentry.rs

```diff
@@ -0,0 +1,712 @@
+//! Transforms the Vercel Log Drain format to Sentry Logs.
+
+use std::str::FromStr;
+
+use chrono::{TimeZone, Utc};
+use relay_conventions::{ENVIRONMENT, ORIGIN, SERVER_ADDRESS, URL_PATH};
+use relay_event_schema::protocol::{Attributes, OurLog, OurLogLevel, SpanId, Timestamp, TraceId};
+use relay_protocol::{Annotated, Meta, Remark, RemarkType};
+use serde::{Deserialize, Serialize};
+
+/// Vercel log structure matching their [schema](https://vercel.com/docs/drains/reference/logs)
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct VercelLog {
+    /// Unique identifier for the log entry.
+    pub id: String,
+    /// Identifier for the Vercel deployment.
+    #[serde(rename = "deploymentId")]
+    pub deployment_id: String,
+    /// Origin of the log.
+    pub source: String,
+    /// Deployment unique URL hostname.
+    pub host: String,
+    /// Unix timestamp in milliseconds.
+    pub timestamp: i64,
+    /// Identifier for the Vercel project.
+    pub project_id: String,
+    // Log severity level.
+    pub level: VercelLogLevel,
+    /// Log message content (may be truncated if over 256 KB).
+    pub message: Option<String>,
+    /// Identifier for the Vercel build. Only present on build logs.
+    pub build_id: Option<String>,
+    /// Entrypoint for the request.
+    pub entrypoint: Option<String>,
+    /// Origin of the external content. Only on external logs.
+    pub destination: Option<String>,
+    /// Function or dynamic path of the request.
+    pub path: Option<String>,
+    /// Log output type.
+    #[serde(rename = "type")]
+    pub log_type: Option<String>,
```

Comment:
nit: This is what we usually name type fields.

```suggestion
    pub ty: Option<String>,
```